src/gluonts/torch/model/deepar/deepar_estimator.py

```diff
@@ -0,0 +1,325 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import List, Optional, Iterable
+
+import numpy as np
+import torch
+from torch.utils.data import DataLoader
+import pytorch_lightning as pl
+
+from gluonts.core.component import validated
+from gluonts.dataset.common import Dataset
+from gluonts.dataset.field_names import FieldName
+from gluonts.itertools import Cyclic, PseudoShuffled, IterableSlice
+from gluonts.time_feature import (
+    TimeFeature,
+    get_lags_for_frequency,
+    time_features_from_frequency_str,
+)
+from gluonts.transform import (
+    Transformation,
+    Chain,
+    RemoveFields,
+    SetField,
+    AsNumpyArray,
+    AddObservedValuesIndicator,
+    AddTimeFeatures,
+    AddAgeFeature,
+    VstackFeatures,
+    InstanceSplitter,
+    ValidationSplitSampler,
+    TestSplitSampler,
+    ExpectedNumInstanceSampler,
+    SelectFields,
+)
+from gluonts.torch.util import (
+    get_forward_input_names,
+    IterableDataset,
+)
+from gluonts.torch.model.estimator import PyTorchLightningEstimator
+from gluonts.torch.model.predictor import PyTorchPredictor
+from gluonts.torch.modules.distribution_output import (
+    DistributionOutput,
+    StudentTOutput,
+)
+
+from .deepar_network import (
+    DeepARLightningNetwork,
+    DeepARNetwork,
+)
+
+
+class DeepAREstimator(PyTorchLightningEstimator):
+    @validated()
+    def __init__(
+        self,
+        freq: str,
+        prediction_length: int,
+        trainer: pl.Trainer = pl.Trainer(),
+        context_length: Optional[int] = None,
+        num_layers: int = 2,
+        num_cells: int = 40,
+        cell_type: str = "LSTM",
+        dropout_rate: float = 0.1,
+        num_feat_dynamic_real: int = 0,
```

Comment:
Yes, I agree. This doesn’t have much to do with Lightning, and is only needed in constructing the network because PyTorch needs to know the input size for the RNN construction (while in PyTorchTS that value is asked directly, I think that asking the size of all different inputs to the `forward` method—and then summing them up—makes more sense from the user perspective).

The “auto mode” you propose makes perfect sense in the estimator. We could just peek into the first (transformed) dataset entry and fill these value up.

Reply:
Could be a follow-up PR, for sure. Just wanted to point it out.